### DISCUSSION TITLE
.NET 9 WASM Threading + Avalonia

### DISCUSSION THREAD

**Discussion by JaggerJo** [AvaloniaUI/Avalonia#17027] [Q&A]
How Production ready is Avalonia for complicated multi threaded apps with .NET 9? 

I've read through the issues in the runtime repo and it seems the threading mode is completely revamped in .NET 9.

https://github.com/dotnet/runtime/issues/85592

**Comment by kekekeks** [comment]
Debugging didn't work last time we've checked it, but we've adapted avalonia to the new mode by spawning a dedicated web-worker for our render thread, so it should work.

**Reply by JaggerJo** [reply]
I've noticed that simply starting the browser project of the xplat template does result in CORS errors because of the new threads as web workers feature.

I'm using `dotnet serve -h "Cross-Origin-Opener-Policy:same-origin" -h "Cross-Origin-Embedder-Policy:require-corp"` instead on the published output.

**Comment by Numpsy** [comment]
fwiw, we were discussing some WASM threading issues over at https://github.com/danipen/TextMateSharp/issues/57 related to syntax highlighting in AvaloniaEdit (where it creates background threads to do some of the work), and it seems to work much better in .NET 9 than in 8 (in 8, it tries to work but is prone to lockups)

**Comment by JaggerJo** [comment]
I've been playing around with `.NET 9 rc1` + `Avalonia 11.2.0-beta2` and got a few demos up and running.

(It required using either chrome or safari technical preview on macOS. )
